alohomora/req.py

```diff
@@ -114,86 +113,172 @@ def __init__(self, idp_url, auth_method=None):
         self.idp_url = idp_url
         self.auth_method = auth_method
 
-    def _validate_u2f_request(self, host, req):
-        LOG.debug('req["appId"]: %s, host: %s', req['appId'], host)
-        return req['appId'] == 'https://%s' % host
+    def _validate_webauthn_request(self, host, appid):
+        LOG.debug('req["appid"]: %s, host: %s', appid, host)
+        return appid == 'https://%s' % host
 
-    def _get_u2f_response(self, reqs):
+    def _get_webauthn_response(self, req):
         """
         Authenticates against yubikey with all the sign requests
         """
-        # if U2F enrolled, requests will look like
-        # [
-        #   {
-        #     "appId": "https://api-12345678.duosecurity.com",
-        #     "challenge": "shfdsjkaKJDGHFSKgfesgfieo2382",
-        #     "keyHandle": "fjdskabghpferwuipgt4iuytr23g4uyiawhbiu",
-        #     "sessionId": "jrt43uiq9tpgh43qu9gbhw3juipgtbw3",
-        #     "version": "U2F_V2"
-        #   },
-        #   { ... }
-        # ]
-        _ = input('Please ensure your security key is plugged in and hit enter...')
-        devices = get_u2f_devices()
-        if not devices:
-            raise IOError('no U2F devices found')
-        LOG.info('U2F requests: %s', reqs)
-        try:
+        # if webauthn enrolled, requests will look like
+        # {
+        #   "extensions": {"appId": "https://api-12345678.duosecurity.com"},
+        #   "challenge": "shfdsjkaKJDGHFSKgfesgfieo2382",
+        #   "allowCredentials": [{
+        #       "transports": ["usb", "nfc", "ble"],
+        #       "type": "public-key",
+        #       "id": "dipB0Q2TgTSpOINIsI9uaesA4ZrI1nGoeKc3Dx-VOvAJ1knOY46MzjY3da14KcTzLPzlIJF9p9gtqr2t6TfWeQ"
+        #   }]
+        #   "sessionId": "dn6WlN9Uunff3ZLSZuu9bdHTr1Nhj0p7Ov89ZcR77nI",
+        #   "userVerification": "discouraged",
+        #   "rpId": "duosecurity.com",
+        #   "timeout": 60000
+        # }
+        def _copy(item):
+            i = {}
+            for k, v in item.items():
+                i[k] = v
+            return i
```

Comment:
Am I missing something? Why not this?
```suggestion
            return item.copy()
```

Reply:
Thanks, dropped that func and used copy directly.